src/components/application_manager/test/commands/mobile/create_interaction_choice_set_test.cc

```diff
@@ -0,0 +1,612 @@
+  command_->Run();
+}
+
+TEST_F(CreateInteractionChoiceSetRequestTest,
+       Run_CheckChoiceSet_InvalidAmountVrCommands_UNSUCCESS) {
+  (*message_)[am::strings::msg_params][am::strings::choice_set][0]
+             [am::strings::menu_name] = kMenuName;
+  (*message_)[am::strings::msg_params][am::strings::choice_set][0]
+             [am::strings::image][am::strings::value] = kImage;
+  (*message_)[am::strings::msg_params][am::strings::choice_set][0]
+             [am::strings::choice_id] = kChoiceId1;
+  (*message_)[am::strings::msg_params][am::strings::choice_set][0]
+             [am::strings::secondary_image][am::strings::value] = kSecondImage;
+
+  FillMessageFieldsItem2(message_);
+  (*message_)[am::strings::msg_params][am::strings::choice_set][1]
+             [am::strings::vr_commands][0] = kVrCommands1;
+  (*message_)[am::strings::msg_params][am::strings::choice_set][1]
+             [am::strings::vr_commands][1] = kVrCommands2;
+
+  EXPECT_CALL(mock_app_manager_, application(_)).WillOnce(Return(app_));
+  EXPECT_CALL(*message_helper_mock_, VerifyImage(_, _, _))
+      .WillRepeatedly(Return(mobile_apis::Result::SUCCESS));
+
+  smart_objects::SmartObject* choice_set_id = NULL;
+  EXPECT_CALL(*app_, FindChoiceSet(kChoiceSetId))
+      .WillOnce(Return(choice_set_id));
+  EXPECT_CALL(mock_app_manager_, GenerateGrammarID()).Times(0);
+  command_->Run();
+}
+
+TEST_F(CreateInteractionChoiceSetRequestTest,
+       Run_IsWhiteSpaceVRCommandsExist_InvalidMenuName_UNSUCCESS) {
+  (*message_)[am::strings::msg_params][am::strings::choice_set][0]
+             [am::strings::menu_name] = "menu_name\t";
+  (*message_)[am::strings::msg_params][am::strings::choice_set][0]
+             [am::strings::secondary_text] = "secondary_text\t";
+  (*message_)[am::strings::msg_params][am::strings::choice_set][0]
+             [am::strings::tertiary_text] = "tertiary_text\t";
+  (*message_)[am::strings::msg_params][am::strings::choice_set][0]
+             [am::strings::image][am::strings::value] = "image\t";
+  (*message_)[am::strings::msg_params][am::strings::choice_set][0]
+             [am::strings::choice_id] = kChoiceId1;
+  (*message_)[am::strings::msg_params][am::strings::choice_set][0]
+             [am::strings::secondary_image][am::strings::value] =
+                 "second_image\t";
+  (*message_)[am::strings::msg_params][am::strings::choice_set][0]
+             [am::strings::vr_commands][0] = "vr_commands_1\t";
+
+  (*message_)[am::strings::msg_params][am::strings::interaction_choice_set_id] =
+      kChoiceSetId;
+
+  smart_objects::SmartObject* choice_set_id = NULL;
+  EXPECT_CALL(*app_, FindChoiceSet(kChoiceSetId))
+      .WillRepeatedly(Return(choice_set_id));
+  EXPECT_CALL(mock_app_manager_, application(_)).WillRepeatedly(Return(app_));
+
+  EXPECT_CALL(*message_helper_mock_, VerifyImage(_, _, _))
+      .WillRepeatedly(Return(mobile_apis::Result::SUCCESS));
+
+  if ((*message_)[am::strings::msg_params][am::strings::choice_set][0]
```

Comment:
@LevchenkoS , yes. This test case tests several posible behaviors. When in different fields we have wrong prompts. So we need statement there is